### PR TITLE
Objects for RP-initiated initiated logout

### PR DESCRIPTION
**Pull request by zachmargolis**
These are the basic objects needed to support RP-initiated logout, I'd like to make sure they look ship-shape before adding methods to use them and follow up in another PR

### CODE REVIEW

File: Source/OIDLogoutResponse.h [hunk 1]
```diff
@@ -0,0 +1,44 @@
+#import <Foundation/Foundation.h>
+
+@class OIDLogoutRequest;
+
+NS_ASSUME_NONNULL_BEGIN
+
+/*! @brief Represents the response to an authorization request.
+ @see https://tools.ietf.org/html/rfc6749#section-4.1.2
```
Comment (WilliamDenniss) [inline Source/OIDLogoutResponse.h]: Comment needs revising

File: Source/OIDLogoutRequest.h [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+}
+
+/*! @brief The service's configuration.
+ @remarks This configuration specifies how to connect to a particular OAuth provider.
+ Configurations may be created manually, or via an OpenID Connect Discovery Document.
+ */
+@property(nonatomic, readonly) OIDServiceConfiguration *configuration;
+
+/*! @brief The client's redirect URI.
+ @remarks post_logout_redirect_uri
```
Comment (WilliamDenniss) [inline Source/OIDLogoutRequest.h]: Indentation is off, here and other comment documentation blocks, please follow style of the rest of the project.

File: Source/OIDLogoutRequest.h [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+
+/*! @brief An opaque value used by the client to maintain state between the request and callback.
+ @remarks state
+ @discussion If this value is not explicitly set, this library will automatically add state and
+ perform appropriate validation of the state in the authorization response. It is recommended
+ that the default implementation of this parameter be used wherever possible. Typically used
+ to prevent CSRF attacks, as recommended in RFC6819 Section 5.3.5.
+ @see https://tools.ietf.org/html/rfc6749#section-4.1.1
+ @see https://tools.ietf.org/html/rfc6819#section-5.3.5
+ @see http://openid.net/specs/openid-connect-session-1_0.html#RPLogout
```
Comment (WilliamDenniss) [inline Source/OIDLogoutRequest.h]: This `@see` link is good (to RP Logout), I think the other two are not related.

File: Source/OIDLogoutRequest.h [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+@property(nonatomic, readonly, nullable) NSString *state;
+
+/*! @brief The client's additional authorization parameters.
+ @see https://tools.ietf.org/html/rfc6749#section-3.1
+ */
+@property(nonatomic, readonly, nullable) NSDictionary<NSString *, NSString *> *additionalParameters;
+
+/*! @internal
+ @brief Unavailable. Please use
+ @c initWithConfiguration:clientId:scopes:redirectURL:additionalParameters:.
```
Comment (WilliamDenniss) [inline Source/OIDLogoutRequest.h]: Needs updating

File: Source/OIDLogoutRequest.m [hunk 5]
```diff
@@ -0,0 +1,173 @@
...
+
+    if (_state) {
+        [query addParameter:kStateKey value:_state];
+    }
+
+    NSAssert(_configuration.discoveryDocument.endSessionEndpoint, OIDMissingEndSessionEndpointMessage);
+
+    // Construct the URL:
+    return [query URLByReplacingQueryInURL:_configuration.discoveryDocument.endSessionEndpoint];
+
```
Comment (WilliamDenniss) [inline Source/OIDLogoutRequest.m]: remove newline

File: Source/OIDLogoutResponse.h [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+@property(nonatomic, readonly, nullable) NSString *state;
+
+/*! @internal
+ @brief Unavailable. Please use initWithParameters:.
+ */
+- (instancetype)init NS_UNAVAILABLE;
+
+/*! @brief Designated initializer.
+ @param request The serviced request.
+ @param parameters The decoded parameters returned from the Authorization Server.
```
Comment (WilliamDenniss) [inline Source/OIDLogoutResponse.h]: Needs revising (not "Authorization Server").

**Comment by WilliamDenniss** [comment]
Hey @zachmargolis, any plans to complete this work?

**Comment by zachmargolis** [comment]
@WilliamDenniss hi, I commented the issue (https://github.com/openid/AppAuth-iOS/issues/132#issuecomment-353249575), I'm no longer working on OIDC stuff right now so I have no plans to complete at this time

**Comment by luksfarris** [comment]
Hi @zachmargolis and @WilliamDenniss , I've created #191 based on this, with fixes for all the comments, as suggested by Zach

**Comment by WilliamDenniss** [comment]
Got it. Closing this in favor of #191.

**Pull request closed without merging**